core/trino-main/src/main/java/io/trino/operator/aggregation/minmaxn/TypedHeap.java

```diff
@@ -111,41 +112,26 @@ public static TypedHeap deserialize(boolean min, MethodHandle compare, Type elem
         return new TypedHeap(min, compare, elementType, capacity, heapBlock.getPositionCount(), heapIndex, heapBlockBuilder);
     }
 
-    public void popAllReverse(BlockBuilder resultBlockBuilder)
+    public void writeAll(BlockBuilder resultBlockBuilder)
     {
         int[] indexes = new int[positionCount];
-        while (positionCount > 0) {
-            indexes[positionCount - 1] = heapIndex[0];
-            positionCount--;
-            heapIndex[0] = heapIndex[positionCount];
-            siftDown();
-        }
+        System.arraycopy(heapIndex, 0, indexes, 0, positionCount);
+        IntArrays.quickSort(indexes, (a, b) -> {
+            try {
+                int comparison = compare(heapBlockBuilder, a, heapBlockBuilder, b);
```

Comment:
If I'm reading this correctly, `compare` doesn't throw, so you should be able to simplify this.  Also, you might want to apply the `min ? comparison : -comparison` in the function so all of this folds to a method reference

Reply:
Ah, yeah, that try-catch was a leftover from an earlier experiment.

> so all of this folds to a method reference

It still needs to reference heapBlockBuilder, so the lambda expression will be there regardless.